aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -3,48 +3,117 @@
 #include "ATen/ExpandUtils.h"
 #include "ATen/NativeFunctions.h"
 #include "ATen/WrapDimUtils.h"
-#include "cpu/UnaryOpsKernel.h"
+
+#include "ATen/CPUApplyUtils.h"
+#include "ATen/Parallel.h"
+#include "ATen/native/cpu/UnaryOpsKernel.h"
 
 #include <algorithm>
+#include <cmath>
 #include <functional>
+#include <iostream>
 #include <numeric>
 #include <vector>
 
 #include <map>
 
 namespace at { namespace native {
 
-#define IMPLEMENT_UNARY_OP(op)                                                \
-Tensor op(const Tensor& self) {                                               \
-  Tensor result = self.type().tensor();                                       \
-  return at::op ## _out(result, self);                                        \
-}                                                                             \
-Tensor& op##_(Tensor& self) {                                                 \
-  return at::op ## _out(self, self);                                          \
-}                                                                             \
-Tensor& _ ## op ## _out_cuda(Tensor& result, const Tensor& self) {            \
-  return at::_ ## op ## _out(result, self);                                   \
-}                                                                             \
-Tensor& _ ## op ## _out_cpu(Tensor& result, const Tensor& self) {             \
-  if (result.is_contiguous() && self.is_contiguous()) {                       \
-    result.resize_(self.sizes());                                             \
-    if (result.numel() > 0) {                                                 \
-      op ## Impl(result, self);                                               \
-    }                                                                         \
-    return result;                                                            \
-  }                                                                           \
-  return at::_ ## op ## _out(result, self);                                   \
+static Tensor _sort_strides(Tensor& tensor_) {
```

Comment:
Ideally this would live in apply utils, so we can benefit from it in more places. The generalization is to take strides of all tensors as tuples, and define an order in which a one tuple is smaller than the other when it's smaller on all coordinates. But this is not a blocker for this PR ofc.

Reply:
For now this still needs to live here, because otherwise we won't get the contiguous case from sorting a transposed contiguous tensors, since we're still branching instead of processing both cases within the CPUApplyUtils.